### PR TITLE
docs: maintainers OEP-55

### PR DESCRIPTION
**Pull request by azanbinzahid**
## Description
This PR updates repo maintainership tools described under [OEP-55 Phase-1](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3426844690/Maintainership+Pilot#Pilot-Phase-1)

## Related Ticket
[PROD-2955](https://2u-internal.atlassian.net/browse/PROD-2955)

### CODE REVIEW

File: catalog-info.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  annotations:
+    # (Optional) Annotation keys and values can be whatever you want.
+    # We use it in Open edX repos to have a comma-separated list of GitHub user
+    # names that might be interested in changes to the architecture of this
+    # component.
+    openedx.org/arch-interest-groups: ""
+spec:
+
+  # (Required) This can be a group(`group:<group_name>` or a user(`user:<github_username>`)
+  owner: user:DawoudSheraz
```
Comment (DawoudSheraz) [inline catalog-info.yaml]: nit: Not me, it would be the Phoenix squad as a whole.

File: catalog-info.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  # (Required) Acceptable Lifecycle Values: experimental, production, deprecated
+  lifecycle: 'production'
+
+  # (Optional) The value can be the name of any known component.
+  subcomponentOf: '<name_of_a_component>'
+
+  # (Optional) An array of different components or resources.
+  dependsOn:
+  - '<component_or_resource>'
+  - '<another_component_or_resource>'
```
Comment (DawoudSheraz) [inline catalog-info.yaml]: lms as well.

File: catalog-info.yaml (hunk 1 above)
Comment (DawoudSheraz) [reply catalog-info.yaml]: Ref https://github.com/openedx/course-discovery/blob/master/openedx.yaml

File: catalog-info.yaml (hunk 2 above)
Comment (azanbinzahid) [reply catalog-info.yaml]: Done

File: catalog-info.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    # component.
+    openedx.org/arch-interest-groups: ""
+spec:
+  owner: group:course-discovery-admins
+  type: 'service'
+  lifecycle: 'production'
+  dependsOn:
+  - studio
+  - taxonomy-connector
+  - ecommerce
```
Comment (DawoudSheraz) [inline catalog-info.yaml]: LMS as well.

File: README.rst [hunk 4]
```diff
@@ -1,29 +1,22 @@
...
-License
--------
+Service providing access to consolidated course and program metadata.
 
-The code in this repository is licensed under version 3 of the AGPL unless otherwise noted. Please see the LICENSE_ file for details.
+Getting Started with Development
+********************************
 
-.. _LICENSE: https://github.com/openedx/course-discovery/blob/master/LICENSE
+This repository works with edx `devstack`_. PLease set it up first and then work under the docker shell
```
Comment (DawoudSheraz) [inline README.rst]: nits: 
- openedx devstack
- PLease -> Please

Comment (DawoudSheraz) [inline README.rst]: We can re-word the 2nd line. Something along the lines of `Once the devstack has been set up and provisioned, run the following commands in devstack directory to access Discovery shell and perform operations as neede`.  Please re-word and add additional context.

**Review by DawoudSheraz — APPROVED** [review]

**Review by ansabgillani — APPROVED** [review]

**Pull request merged**